contracts/math/DecimalMath.sol

```diff
@@ -17,37 +19,66 @@ library DecimalMath {
         return x.add(y);
     }
 
+    /// @dev Adds x and y, assuming they are both fixed point with 18 decimals.
+    function addd(int256 x, int256 y) internal pure returns (int256) {
+        return x.add(y);
+    }
+
     /// @dev Substracts y from x, assuming they are both fixed point with 18 decimals.
     function subd(uint256 x, uint256 y) internal pure returns (uint256) {
         return x.sub(y);
     }
 
+    /// @dev Substracts y from x, assuming they are both fixed point with 18 decimals.
+    function subd(int256 x, int256 y) internal pure returns (int256) {
+        return x.sub(y);
+    }
+
     /// @dev Multiplies x and y, assuming they are both fixed point with 18 digits.
     function muld(uint256 x, uint256 y) internal pure returns (uint256) {
         return muld(x, y, 18);
     }
 
+    /// @dev Multiplies x and y, assuming they are both fixed point with 18 digits.
+    function muld(int256 x, int256 y) internal pure returns (int256) {
+        return muld(x, y, 18);
+    }
+
     /// @dev Multiplies x and y, assuming they are both fixed point with `decimals` digits.
-    function muld(
-        uint256 x,
-        uint256 y,
-        uint8 decimals
-    ) internal pure returns (uint256) {
+    function muld(uint256 x, uint256 y, uint8 decimals)
+        internal pure returns (uint256)
+    {
         return x.mul(y).div(unit(decimals));
     }
 
+    /// @dev Multiplies x and y, assuming they are both fixed point with `decimals` digits.
+    function muld(int256 x, int256 y, uint8 decimals)
+        internal pure returns (int256)
+    {
+        return x.mul(y).div(int256(unit(decimals)));
+    }
+
     /// @dev Divides x between y, assuming they are both fixed point with 18 digits.
     function divd(uint256 x, uint256 y) internal pure returns (uint256) {
         return divd(x, y, 18);
     }
 
+    /// @dev Divides x between y, assuming they are both fixed point with 18 digits.
+    function divd(int256 x, int256 y) internal pure returns (int256) {
+        return divd(x, y, 18);
```

Comment:
What if I **4/(-4)**?

Reply:
That should return -1, why?

Reply:
Ah sorry, I did miss one thing. Yeah. Ok. Looks interesting indeed. Awesome work :clap: 
It's already approved BTW.